aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -530,22 +588,24 @@ void index_add_cuda_impl(const Tensor& self, int64_t dim, const Tensor& index, c
 
   int mpc = at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
 
-#define SMALL_INDEX(TENSOR_TYPE, INDICES_TYPE, TYPE, SELF_DIM, SOURCE_DIM, IDX_DIM)  \
-  indexAddSmallIndex<TENSOR_TYPE, INDICES_TYPE, TYPE, SELF_DIM, SOURCE_DIM, IDX_DIM> \
-    <<<smallIndexGrid, smallIndexBlock, 0, stream>>>(                                \
-      selfInfo, sourceInfo, indexInfo,                                               \
-      selfAddDim, sourceAddDim, sliceSize, selfAddDimSize, alpha_value);             \
+#define SMALL_INDEX(TENSOR_TYPE, INDICES_TYPE, TYPE, SELF_DIM, SOURCE_DIM, IDX_DIM)     \
+  indexFuncSmallIndex<TENSOR_TYPE, INDICES_TYPE, TYPE, SELF_DIM, SOURCE_DIM, IDX_DIM,   \
+                      /*func_t=*/ReduceAdd, /*use_alpha=*/true>                         \
```

Comment:
if you remove `use_alpha` template parameter, you won't have to specify ReduceAdd here, it will be inferred from the type of `reduce_add` argument